pkg/daemon/ceph/client/status.go

```diff
@@ -217,15 +232,15 @@ func IsClusterCleanError(context *clusterd.Context, clusterInfo *ClusterInfo) er
 	return nil
 }
 
-func isClusterClean(status CephStatus) (string, bool) {
+func isClusterClean(status CephStatus, pgHealthyRegex *regexp.Regexp) (string, bool) {
 	if status.PgMap.NumPgs == 0 {
 		// there are no PGs yet, that still counts as clean
 		return "cluster has no PGs", true
 	}
 
 	cleanPGs := 0
 	for _, pg := range status.PgMap.PgsByState {
-		if pg.StateName == activeClean || pg.StateName == activeCleanScrubbing || pg.StateName == activeCleanScrubbingDeep {
```

Comment:
is there any problem in checking this way, I feel like if there is ceph API change we need to re-visit the regex again

Reply:
Indeed, we need to fix `defaultPgHealthyRegex` if the Ceph's specification of the state names changes (e.g., the state names might be concatenated with `-` instead of `+` in the future). However, in this case, we need to fix this conditional expression anyway, even without my PR. I don't think this PR adds to the hassle.

As the original issue says, this PR allows users to change the condition under which Rook updates the PDBs. For example, a user could use this feature to reset the PDBs even during remapping. In this case, the user will need to fix their `pgHealthyRegex` if the specification of the Ceph's state names changes.

Reply:
okay thanks for the explanation, @sp98 do you agree with it